src/qt/pivx/topbar.cpp

```diff
@@ -251,10 +251,19 @@ void TopBar::lockDropdownMouseLeave(){
 
 void TopBar::onBtnReceiveClicked(){
     if(walletModel) {
+        QString addressStr = walletModel->getAddressTableModel()->getLastUnusedAddress();
+        if (addressStr.isNull()) {
+            // For some reason we don't have any address in our address book, let's create one
+            CBitcoinAddress newAddress;
+            if (!walletModel->getNewAddress(newAddress, "Default").result) {
+                inform(tr("Error generating address"));
+                return;
+            }
+            addressStr = QString::fromStdString(newAddress.ToString());
+        }
```

Comment:
I think it would make more sense, when there's no address, to create the new one directly from the addressTableModel, or walletModel (rather than from TopBar).

Reply:
Context, creation time?

The wallet UI is creating a "default" address at creation time (when it's not encrypted). 

This `getNewAddress` call could only be called if there was an issue with the dat file, if the user/software for whatever reason screwed up the file and there is no stored addresses.

In every other scenario, there will be at least one address created on the address book.

Reply:
I mean just moving this logic elsewhere, e.g. in getLastUnusedAddress

Reply:
ah ok, yeah.

Reply:
done